### PR TITLE
Make ConfigurationManager.Sources public

### PR DESCRIPTION
**Pull request by fredjeck**
ConfigurationManager.Sources is now a public property rather than
an explicit interface implementation of  ConfigurationBuilder.Sources.

#61675

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66485) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66485) before we can accept your contribution.<br/><br/>:x: fredjeck [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66485)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=66485) it.</sub>

File: src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs [hunk 1]
```diff
@@ -66,7 +66,8 @@ public ConfigurationManager()
 
         IDictionary<string, object> IConfigurationBuilder.Properties => _properties;
 
-        IList<IConfigurationSource> IConfigurationBuilder.Sources => _sources;
+        /// <inheritdoc />
+        public IList<IConfigurationSource> Sources => _sources;
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs:70]: Can you add a test to ensure clearing works as expected? That was the scenario reported in the original issue.

Comment (fredjeck) [reply src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs:70]: That's now done

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for the contribution!

**Review by maryamariyan — APPROVED** [review]

**Pull request merged**